packages/next/src/build/handle-externals.ts

```diff
@@ -130,6 +141,11 @@ export async function resolveExternal(
   return { res, isEsm }
 }
 
+function getPackageName(filePath: string) {
+  const match = filePath.match(/node_modules(?:\/\.pnpm\/[^/]+)?\/([^/]+)/)
```

Comment:
Please include some examples here. Regular expressions are notoriously hard to debug.

Reply:
extracted as a lib and add some tests